### PR TITLE
feat: don't require .env to exist

### PR DESCRIPTION
**Pull request by odilitime**
# Risks

Medium, haven't discussed this widely

# Background

## What does this PR do?

Removes the requirement for .env to exists

## What kind of change is this?

Improvement

## Why are we doing this? Any context or related work?

Large multiagent instances will have to rely on character.json to have all the settings and secrets for that agent. Have a default .env in these instances to set defaults might be helpful but also hurtful. This allows the admin to choose if they want any defaults across all their agents or nots.

# Documentation changes needed?

My changes do not require a change to the project documentation.

# Testing

Ran against stable-11-17 branch without issue

### CODE REVIEW

File: packages/core/src/settings.ts [hunk 1]
```diff
@@ -69,18 +69,12 @@ export function loadEnvConfig(): Settings {
...
     const envPath = findNearestEnvFile();
 
-    if (!envPath) {
-        throw new Error("No .env file found in current or parent directories.");
-    }
-
-    // Load the .env file
-    const result = config({ path: envPath });
+    // attempt to Load the .env file into process.env
+    const result = config(envPath ? { path: envPath } : {});
```
Comment (monilpat) [inline packages/core/src/settings.ts:73]: Confirming that config({}) doesn't cause any errors?

**Comment by monilpat** [comment]
FYI: There is another approach to agent specific envs added in: https://github.com/ai16z/eliza/pull/410 still via env but in a segmented way.

File: packages/core/src/settings.ts (hunk 1 above)
Comment (odilitime) [reply packages/core/src/settings.ts:73]: It does not, I checked

**Comment by odilitime** [comment]
I saw that, complementary to this. Wasn't sure if that's going to get merged so left it out of this description

**Comment by monilpat** [comment]
Awesome - re comments. Yeah totally makes sense. Feel free to add a quick
note in the README about .env files being optional. I know it may cause
merge conflicts between these 2 PRs, but it does seem important to have!
Thanks for doing this :)

**Comment by monilpat** [comment]
Looks like the limitations are: feat|fix|docs|style|refactor|test|chore:any description see

 ```
 if [[ ! "$PR_TITLE" =~ ^(feat|fix|docs|style|refactor|test|chore):\ .+ ]]; then
                    echo "PR title does not match the required pattern."
                    exit 1
                  fi
```

**Comment by monilpat** [comment]
LGTM! 🥇

**Pull request merged**